Update homepage sections and spacing for new design

MartinJJones opened the pull request:
## What

Create new view templates and stylesheets that are used when `new_design` is true, when requested the layout is updated so that:

- The "More on GOV.UK" is moved next to "Government Activity"
  - Update the styling of the "More on GOV.UK" section so the links appear in a single column
- Section heading font-sizes are increased from `m` to `l`
- Section spacing is updated so that spacing for the top and bottom of each section adds up to 50px on mobile and 60px on desktop
- Removed "Departments and organisations" section
- Updated `index_section_count` and `indexes_section` values are also updated to reflect the new position on the page, this will ensure correct reporting in Google Analytics
- Updated the Chevron card styling so that all content is vertically centred and the top border is removed from the first chevron-card element

## Why

This is part of changes that will be gradually rolled out to the homepage. As these changes are not going to be made live immediately, I have attempted to split the code in a way that will allow them to be easily toggled by the new_design parameter.

[Trello card](https://trello.com/c/5AdPoKbt/2060-look-and-feel-homepage-general-adjustments-m), [Jira issue NAV-8491](https://gov-uk.atlassian.net/browse/NAV-8491)

## Visual Changes

### Sections

#### Before
<img width="1092" alt="bottom-section-before" src="https://github.com/alphagov/frontend/assets/28779939/a8656575-3de0-4e73-91de-0a67f1d6b57c">

#### After
<img width="1011" alt="bottom-section-after" src="https://github.com/alphagov/frontend/assets/28779939/ad8d25c7-56c8-4599-9727-64f8615c6978">

### Chevron cards

#### Before
<img width="634" alt="chevron-card-before" src="https://github.com/alphagov/frontend/assets/28779939/772d9cb9-8cfa-465a-a4d9-4f64f47f311e">

#### After
<img width="629" alt="chevron-card-after" src="https://github.com/alphagov/frontend/assets/28779939/904a1939-d495-49c3-a52b-d9487afcf7ba">

---
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

MartinJJones committed:
Move "More on GOV.UK" section

- Use the `new_design?` method to load the required view templates
- For the new design, "More on GOV.UK" has moved next to "Government Activity"
- The index_section_count and indexes_section values are also updated to reflect the new position on the page

MartinJJones committed:
Update styling for "More on GOV.UK" section

- Ensure the section is using the grid system
- Update the styling of the "More on GOV.UK" section so the links appear in a single column

MartinJJones committed:
Increase heading sizes on homepage if new_design is true

MartinJJones committed:
Use `homepage-section__heading` for promo section

- Create a new `promotion_slots_new` view template
- Increase font-size to "l"
- Wrap the heading in a `homepage-section__heading` class, this matches the approach used in other sections

MartinJJones committed:
Create _homepage_new.scss for new design

- This will be used to update the styles on all sections of the homepage
- Will be requested when `new_design?` is true

MartinJJones committed:
Update sections on homepage if new_design is true

If new_design is true:

- "Departments and organisations" is removed
- "More on GOV.UK" is moved next to Gov activity
- A new blue border style is applied to separate the sections
- Ensure that the home-section__heading border is not visible on desktop
- Remove the border-top styling specific to the promotion-slots heading
- Group all new sections together in the view template

MartinJJones committed:
Update section spacing if new_design is true

If new_design is true, then all of the sections on the homepage will use the new spacing, the spacing for the top and bottom of each section needs to be 50px on mobile and 60px on desktop.

MartinJJones committed:
Update chevron-card styles for new homepage design

The styling of the chevron-card has been updated for the new homepage design to ensure that:

- all content and the chevron icon is vertically centred
- the top border is removed from the first element in the list

MartinJJones commented:
@andysellick would you also be able to help review this PR? thinking specifically about the updates to the sections on the homepage and the related `index_section` and `index_section_count` for Google Analytics when using the new design - https://github.com/alphagov/frontend/pull/3796/files#diff-b0533c1f365ffe916d27cb02d89deee3e7e1d586f289692af1c69b6036d88449

andysellick commented on app/views/homepage/index.html.erb:
```diff
@@ -13,7 +17,11 @@
+      index_section_count = 5
+    else
+      index_section_count = 6
+    end
```
I think you could also write this as

```Ruby
index_section_count = 6
index_section_count = 5 if new_design?
```

MartinJJones replied:
Thanks Andy, this looks much cleaner, I've updated the view template to use this approach 👍

andysellick reviewed (commented):
Index stuff looks good to me 👍

patrickpatrickpatrick commented:
This is a minor accessibility thing, I'm wondering if [clickable area/touch target of the chevron cards should be adjusted so there's some space between them](https://www.w3.org/TR/mobile-accessibility-mapping/#touch-target-size-and-spacing). They have a pseudo element applied to them which makes the whole element clickable but if you are using zoom it might make distinguishing between the cards difficult.

![Screenshot 2023-10-09 at 14 19 42](https://github.com/alphagov/frontend/assets/3727504/e68e9ff1-133d-4ce6-ab1d-cfbe39fe5ab6)

MartinJJones committed:
Ensure tap target sizes remain accessible

MartinJJones commented:
> This is a minor accessibility thing, I'm wondering if [clickable area/touch target of the chevron cards should be adjusted so there's some space between them](https://www.w3.org/TR/mobile-accessibility-mapping/#touch-target-size-and-spacing). They have a pseudo element applied to them which makes the whole element clickable but if you are using zoom it might make distinguishing between the cards difficult.
> 
> ![Screenshot 2023-10-09 at 14 19 42](https://user-images.githubusercontent.com/3727504/273597344-e68e9ff1-133d-4ce6-ab1d-cfbe39fe5ab6.png)

Thanks @patrickpatrickpatrick good spot!, I've updated the padding in a recent commit to ensure that the target size for the chevron-card is the same as before, and updated the padding on `chevron-card__wrapper` to ensure the total padding size is 24px at the top and bottom as needed in the design.

MartinJJones committed:
Update government activity description and spacing

- Change the lead_paragraph text to regular use a regular paragraph tag with a `govuk-body` class
- Add margin bottom to the "Government activity" heading to add some space before the description

MartinJJones commented:
> I think the spacing here
> 
> ![Screenshot 2023-10-09 at 14 10 33](https://user-images.githubusercontent.com/3727504/273595202-061732ab-4e77-48df-95c0-8d2497ec85bb.png)
> 
> isn't the same as in the design https://www.figma.com/file/InXsSt89C5y5QRT1DZ3dPM/Homepage-redesign---2023?type=design&node-id=2522-77874&mode=design&t=7uUIoRtetLK4e6ac-0

Thanks @patrickpatrickpatrick I've updated this section so it now closely matches the Figma design

patrickpatrickpatrick commented:
Looks good to me now, thanks Martin!
